include/dlaf/common/round_robin.h

```diff
@@ -0,0 +1,36 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <vector>
+
+namespace dlaf {
+namespace common {
+
+template <class T>
+struct RoundRobin {
+  template <class... Args>
+  RoundRobin(std::size_t n, Args... args) : curr_index_(0) {
```

Comment:
@teonnik I also switched to passing `args` by value instead of universal reference since we are going to use it multiple times (for constructing the n resources) and it may be problematic if we pass in a movable argument.

Reply:
I would also add a `std::move` to potentially help the compiler a little bit : `pool_.emplace_back(std::move(args)...);`

Reply:
Note that you can still take them by universal reference, as long as you don't forward them in the `emplace_back` call.